### PR TITLE
Call getpwuid only after chroot is set

### PR DESCRIPTION
**Pull request by zedi-pramodh**
There is a regression in recent commit where getpwuid() was added before chroot is set. Due to that any container which runs with non-root userid will fail to start. 

Testing done
===========
Before this fix grafana container which runs with user id 472 was failing to start. It works with this fix.

### CODE REVIEW

**Comment by zedi-pramodh** [comment]
Yes this need to be back ported. I labeled as stable.

**Review by rouming — COMMENTED** [review]
The original intention of the change was to be sure shim vm has the same user, but let @shjala review that.

My only request is please add a proper commit message describing the problem and what exact commit this fix targets. Thanks.

**Comment by zedi-pramodh** [comment]
Updated the commit message

**Comment by zedi-pramodh** [comment]
> What is the grafana container mentioned in the PR description? I've tried to find it in the scripts and cannot find it... Should it even use this util? I think the use case should be discussed with @shjala.
> 
> Until now the only use case that I'm aware about is here:

I did not understand your comment. Customers can use any container image, grafana is just an app running in the container. It happens that it runs with non-root userid. You can check alpha for that image if you are interested.

**Comment by OhmSpectator** [comment]
>  Customers can use any container image, grafana is just an app running in the container.
So, is it just a regular container Applicaion deployed on a node?

**Comment by zedi-pramodh** [comment]
> So, is it just a regular container Application deployed on a node?

Yes.

**Comment by OhmSpectator** [comment]
So, it means EVE has been failing to run any container App since 11.3.0... Wow.
It's good that we do not have the original code in any stable/LTS branch, so no customer is affected, and the backports are not needed.

**Comment by OhmSpectator** [comment]
Another question: why our regression tests did not detect this problem... @uncleDecart, do you know whether we have any tests for the container-based apps?

**Comment by zedi-pramodh** [comment]
Its not all containers fail to start. Only containers that run the apps with non-root user. May be most of the test cases run the apps in the container as root user.

**Review by eriknordmark — APPROVED** [review]
Kick off tests; @shjala should review before we merge this.

**Comment by milan-zededa** [comment]
> Another question: why our regression tests did not detect this problem... @uncleDecart, do you know whether we have any tests for the container-based apps?

@OhmSpectator It was detected but Shah is busy this week with high-priority TPM issues: https://github.com/lf-edge/eden/issues/944

**Comment by eriknordmark** [comment]
Based on the discussion in https://github.com/lf-edge/eden/issues/944 this fix is needed.
(And we need to get the eden tests to pass regularly to not miss a test failure like this in the noise of other stochastic failures.)

**Pull request merged**
